src/Raven.Server/Documents/Handlers/Processors/Revisions/RevisionsHandlerProcessorForRevertRevisions.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using JetBrains.Annotations;
+using Raven.Server.Documents.Operations;
+using Raven.Server.Documents.Revisions;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+
+namespace Raven.Server.Documents.Handlers.Processors.Revisions
+{
+    internal class RevisionsHandlerProcessorForRevertRevisions : AbstractRevisionsHandlerProcessorForRevertRevisions<DatabaseRequestHandler, DocumentsOperationContext>
+    {
+        public RevisionsHandlerProcessorForRevertRevisions([NotNull] DatabaseRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+
+        protected override void RevertRevisions(long operationId, RevertRevisionsRequest configuration, OperationCancelToken token)
+        {
+            var t = RequestHandler.Database.Operations.AddLocalOperation(
+                operationId,
+                OperationType.DatabaseRevert,
+                $"Revert database '{RequestHandler.Database.Name}' to {configuration.Time} UTC.",
+                detailedDescription: null,
+                onProgress => RequestHandler.Database.DocumentsStorage.RevisionsStorage.RevertRevisions(configuration.Time, TimeSpan.FromSeconds(configuration.WindowInSec), onProgress, token),
+                token: token);
+        }
+
+        protected override long GetNextOperationId()
+        {
+            return ServerStore.Operations.GetNextOperationId();
```

Comment:
you want to get the operation Id from the RequestHandler.Database.Operations.GetNextOperationId(), not from the server one.

Reply:
I see it was like that in the old code, this is a mistake. We should fix it in older versions as well.

Reply:
done